### PR TITLE
414 clear stale credentials

### LINKED ISSUE #414: automatic clearing of stale credentials from kanidm_tokens
`~/.cache/kanidm_tokens` gets full of all the used tokens, is there an auto-cleanup task, or could there be one made?

Two reasons:

 1. security - it's not good to list what accounts have been used on this machine by this account - `user_a` has access to `idm_admin` - rad, target them!
 2. it's neater. uWu.

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #414 - this allows cleaning up of stale sessions, as well as show session details, and doing a "logout" of sessions. 

Note - this isn't finished yet, as the current integration with bundy and how we sign/generate UAT's needs to be reworked. I'll update the PR with that shortly. 

- [ x ] cargo fmt has been run
- [x] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ x ] book chapter included (if relevant)
- [ - ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
This is now complete, and ready for review. Sorry, it's quite large :)

File: kanidmd/src/lib/core/https.rs [hunk 1]
```diff
@@ -1155,30 +1149,32 @@ pub fn create_https_server(
...
     qe_w_ref: &'static QueryServerWriteV1,
     qe_r_ref: &'static QueryServerReadV1,
 ) -> Result<(), ()> {
     info!("WEB_UI_PKG_PATH -> {}", env!("KANIDM_WEB_UI_PKG_PATH"));
 
-    // Create the in memory fernet key
-    let fernet_handle = fernet::Fernet::new(&fernet::Fernet::generate_key()).ok_or_else(|| {
-        error!("Failed to generate fernet key");
+    let bundy_handle = bundy::hs512::HS512::from_str(bundy_key).map_err(|e| {
+        error!("Failed to generate bundy handle - {:?}", e);
```
Comment (yaleman) [inline kanidmd/src/lib/core/https.rs:1160]: This is going to come up as  "what does this even mean?!?!?!" ticket in about .. a year. :)

**Review by yaleman — CHANGES_REQUESTED** [review]
All the changes look sane.

From a usage perspective, given this'll require reinstalling client and server tooling, let's bump the version ;) 

https://github.com/kanidm/kanidm/pull/447/files#diff-c84d80f3026964fe4f44d89fb4f61ba7d8d8df326fa5a9dfe4ccb02c91d50474R3

**Comment by Firstyear** [comment]
> 
> From a usage perspective, given this'll require reinstalling client and server tooling, let's bump the version ;)
> 

The version bumps when I do the releases - the next one is 1st of July. There is some automated tooling for this, so I normally don't touch the version in the Cargo.toml until then because else it gets confused and sad.

File: kanidmd/src/lib/core/https.rs (hunk 1 above)
Comment (Firstyear) [reply kanidmd/src/lib/core/https.rs:1160]: If this fails to generate there is going to be some "chaos" 😅

**Review by victorcwai — APPROVED** [review]
Overall I think it looks great! I don't see any obvious error at least.

**Comment by Firstyear** [comment]
@victorcwai Great! Did you have any questions about how it works? Or happy for me to merge it?

**Comment by victorcwai** [comment]
I can see how the UAT is updated to enable showing session details to client and doing logout. Using bundy to sign, verifying if the token has timed out, etc. So I understand it logically and the code seemed fine to me. 
I am happy and you can merge it!

**Pull request merged**